csharp/src/Ice/WSTransceiver.cs

```diff
@@ -1462,36 +1451,30 @@ private bool PostWrite(int size, ref int offset, int status)
 
         private bool ReadBuffered(int sz)
         {
-            if (_readBufferPos == _readBuffer.B.Position())
+            if (_readBufferPos == _readBufferOffset)
             {
-                _readBuffer.Resize(_readBufferSize, true);
+                _readBuffer = new byte[_readBufferSize];
```

Comment:
Why do we allocate a new buffer here? Can't we just re-use the previous array?

Reply:
Wasn't the previous code calling Resize, to avoid holding a large buffer and going back to a buffer equal to `_readBufferSize`, I guess it will be more correct to reuse it if size equal to `_readBufferSize`?

Reply:
`Buffer.Resize` never shrinks the buffer.  I don't think the `_readBuffer` will ever grow to something large because it's mostly used to read the WS frame header, WS control frames (which are guaranteed to have a size < 125) and the start of the data payload, not the full data payload (the reminder of the data payload is read directly in the provided buffer).

Reply:
https://github.com/zeroc-ice/ice/blob/master/csharp/src/Ice/WSTransceiver.cs#L1483

Doesn't this assumes `_readBuffer` can grow more than `_readBufferSize`

Reply:
Yes, but if you check the callers you'll see that the only case where `sz` can be superior to `_readBufferSize` is eventually for the control frame payload but according to the spec this payload should be small (the frame size should be < 125...). So my guess is that we did this just to be on the safe side...

For 4.0, we could get rid of the resize there and be strict by adding an assert with a comment explaining that `sz` should always be smaller than `_readBufferSize` because a control frame size is always < 125 (https://tools.ietf.org/html/rfc6455#section-5.5).

Reply:
Reworked ReadBuffered to reuse the read buffer